src/main/java/seedu/sellsavvy/logic/commands/ordercommands/AddOrderCommand.java

```diff
@@ -1,17 +1,99 @@
 package seedu.sellsavvy.logic.commands.ordercommands;
 
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_COUNT;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_ITEM;
+
+import java.util.List;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.commons.util.ToStringBuilder;
+import seedu.sellsavvy.logic.Messages;
 import seedu.sellsavvy.logic.commands.Command;
 import seedu.sellsavvy.logic.commands.CommandResult;
 import seedu.sellsavvy.logic.commands.exceptions.CommandException;
 import seedu.sellsavvy.model.Model;
+import seedu.sellsavvy.model.order.Order;
+import seedu.sellsavvy.model.order.OrderList;
+import seedu.sellsavvy.model.person.Person;
 
 /**
  * Adds an order under a specified person.
  */
 public class AddOrderCommand extends Command {
 
+    public static final String COMMAND_WORD = "addOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an order under the specified person. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_ITEM + "ITEM "
+            + PREFIX_DATE + "DELIVERY_BY "
+            + PREFIX_COUNT + "ITEM_COUNT\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_ITEM + "Lamp "
+            + PREFIX_DATE + "20-12-2024 "
+            + PREFIX_COUNT + "2";
+
+    public static final String MESSAGE_SUCCESS = "New Order added for %1$s: %2$s";
+    public static final String MESSAGE_DUPLICATE_WARNING = "Note: "
+            + "This customer already has an order for this item"
+            + ", verify if this is a mistake\n";
+
+    private final Index index;
+    private final Order toAdd;
+
+    /**
+     * Creates an AddOrderCommand to add the specific order under
+     */
+    public AddOrderCommand(Index index, Order order) {
+        requireAllNonNull(index, order);
+        this.index = index;
+        this.toAdd = order;
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return null; // to be implemented
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToAddUnder = lastShownList.get(index.getZeroBased());
+        OrderList orderList = personToAddUnder.getOrderList();
+        String feedbackToUser = orderList.contains(toAdd) ? MESSAGE_DUPLICATE_WARNING : "";
```

Comment:
Just a KIV. In future when we add ability to mark order, we should only care about those unmarked one as it is totally fine for customer to order the same thing again if they like what they recieved last time. By then we need change the MESSAGE_DUPLICATE_WARNING as well.

Reply:
honestly i also don't know but we should definitely only considered whether it is not completed but the other variables can be quite controversial. I never think that far yet maybe we discussed it in our next meeting.

Reply:
Maybe can create an issue about this first?

Reply:
ok